### PR TITLE
Store everything in the fheroes2.cfg, remove the binary config file support

### LINKED ISSUE #6161: Rework of experimental options
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

Since there are always ongoing debates about whether to keep or remove certain experimental options we need to make a clear way how to deal with them.

### Describe the possible solution

I suggest to do these steps:
- [ ] add a configuration option to enable and disable all experimental options in the game. With the disabled state there is no way to change them or even load them with save files (controversial)
- [ ] save them in a separate file format which supports extension in the future
- [ ] handle issues with options being loaded from save files which update global state of these options within the game
- [ ] add a clear indicator that the game is being played with these options being enabled (an icon perhaps on the Adventure Map)
- [ ] propose a new dialog for these options to keep the same style of the game. Each option should follow the guide from [the wiki page](https://github.com/ihhub/fheroes2/wiki/Adding-a-new-option-into-the-game). We might stick to one set of sprites to turning on or off. Alternatively, we can use checkboxes but popup windows with a full description must be present. The page must contain a message telling that these options change the core gameplay and can lead to unexpected consequences.
- [ ] add an indicator that a save file contains enabled experimental options
- [ ] add an indicator that a high score was achieved with the enabled options
- [ ] add a question dialog for users which do not have these options enabled (global configuration flag) while they are loading a save file with the options
- [ ] add a way to show the options for campaign scenarios
- [ ] rename experimental options into something more game-style friendly

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
close #6161

This should be merged after #6239 and #6192/#6195, so we need to make a decision regarding the last two :)

This PR also adds a mechanism for storing window coordinates in "no interface" mode in `fheroes2.cfg`. `fheroes2.bin` is not needed anymore and may be removed.

### CODE REVIEW

**Review by ihhub — APPROVED** [review]

**Review by Branikolog — APPROVED** [review]

**Comment by ihhub** [comment]
@oleg-derevenetz , great thanks for handling most of experimental options work!

File: src/engine/tinyconfig.cpp [hunk 1]
```diff
@@ -95,6 +103,44 @@ std::string TinyConfig::StrParams( const std::string & key ) const
...
+
+    auto convertToInt = []( const std::string & str, auto & intValue ) {
+        const auto [ptr, ec] = std::from_chars( str.data(), str.data() + str.size(), intValue );
+
+        return ec == std::errc();
+    };
+
+    fheroes2::Point result;
+
+    static_assert( std::is_integral<decltype( result.x )>::value && std::is_integral<decltype( result.y )>::value,
```
[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/engine/tinyconfig.cpp:131]: @ihhub Stupid SonarCloud now [reports](https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=BUG&id=ihhub_fheroes2&open=AYSzblF-qAXo8N3jxFqe) this as a "bug", while it is obviously a false positive. Apparently this analysis doesn't work well enough with static compile-time checks.

[the remark this answers is not in the document]
Comment (ihhub) [reply src/engine/tinyconfig.cpp:131]: Hi @oleg-derevenetz , `PointBase2D` structure has only one template parameters so check x or y would be the same.

[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/engine/tinyconfig.cpp:131]: @ihhub This is the case now, but this check is designed to protect against future changes, and we don't know their nature right now, so both fields are checked.

**Pull request merged**